eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,25 +312,63 @@ foreach ($templateFile in $templateFiles) {
         }
     }
 
-    foreach ($key in $deploymentOutputs.Keys)
+    if ($OutFile)
     {
-        $value = $deploymentOutputs[$key]
-        
-        if ($CI) {
-            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-            Write-Host "Setting variable '$key': ***"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
-            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
-        } else {
-            Write-Host ($shellExportFormat -f $key, $value)
+        if (!$IsWindows)
+        {
+            Write-Host "File option is supported only on Windows"
         }
+
+        # Compute a hash of repository root to scope the test environment file
+        $repositoryRootHash = New-Object System.Text.StringBuilder
+        [System.Security.Cryptography.HashAlgorithm]::Create("SHA1").ComputeHash([System.Text.Encoding]::UTF8.GetBytes($repositoryRoot)) | `
+            %{ [Void]$repositoryRootHash.Append($_.ToString("x2")) }
+
+        $outputPath = Join-Path $env:USERPROFILE ".Azure" "TestEnvironments" $repositoryRootHash
+        $environmentInfoFile =  Join-Path $outputPath "environment.txt"
+
+        # Write an environment info file
+        Set-Content $environmentInfoFile -Value "Test environments for $repositoryRoot repository" -Force
+
+        $outputFile = Join-Path $outputPath $serviceName
+
```

Comment:
Just for context it is probably still worth dumping the keys for the values to the console so folks know what was written to the file.

Reply:
The `
        Write-Host "Test environment settings`n $environmentText`nstored into encrypted $outputFile"` line does it.